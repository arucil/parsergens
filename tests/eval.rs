
use insta::assert_debug_snapshot;

parsergens::parsergen!(mod clr_parser: LR("fixtures/expr_eval.pg"));

#[test]
fn clr() {
  let result = clr_parser::with_input(r"(3.2 * 51 + 1) / 20     ")
    .parse_expr();
  assert_debug_snapshot!(result);
}

parsergens::parsergen!(mod lalr_parser: LALR("fixtures/expr_eval.pg"));

#[test]
fn lalr() {
  let result = lalr_parser::with_input(r"(3.2 * 51 + 1) / 20     ")
    .parse_expr();
  assert_debug_snapshot!(result);
}
