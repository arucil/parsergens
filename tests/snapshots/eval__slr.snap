---
source: tests/eval.rs
expression: tokens
---
Ok(
    Ok(
        8.21,
    ),
)
